plugins/toolkit/jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/amazonqFeatureDev/util/FeatureDevClientUtilTest.kt

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.codewhisperer.amazonqFeatureDev.util
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.testFramework.RuleChain
+import com.intellij.testFramework.replaceService
+import kotlinx.coroutines.test.runTest
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+import software.amazon.awssdk.awscore.exception.AwsErrorDetails
+import software.amazon.awssdk.services.codewhispererruntime.model.CreateUploadUrlResponse
+import software.amazon.awssdk.services.codewhispererruntime.model.ValidationException
+import software.amazon.awssdk.services.codewhispererstreaming.model.ThrottlingException
+import software.aws.toolkits.core.telemetry.TelemetryBatcher
+import software.aws.toolkits.core.telemetry.TelemetryPublisher
+import software.aws.toolkits.jetbrains.services.amazonqFeatureDev.ContentLengthError
+import software.aws.toolkits.jetbrains.services.amazonqFeatureDev.PlanIterationLimitError
+import software.aws.toolkits.jetbrains.services.amazonqFeatureDev.clients.FeatureDevClient
+import software.aws.toolkits.jetbrains.services.amazonqFeatureDev.clients.GenerateTaskAssistPlanResult
+import software.aws.toolkits.jetbrains.services.amazonqFeatureDev.util.createConversation
+import software.aws.toolkits.jetbrains.services.amazonqFeatureDev.util.createUploadUrl
+import software.aws.toolkits.jetbrains.services.amazonqFeatureDev.util.generatePlan
+import software.aws.toolkits.jetbrains.services.codewhisperer.CodeWhispererTestUtil
+import software.aws.toolkits.jetbrains.services.codewhisperer.amazonqFeatureDev.FeatureDevTestBase
+import software.aws.toolkits.jetbrains.services.telemetry.NoOpPublisher
+import software.aws.toolkits.jetbrains.services.telemetry.TelemetryService
+import software.aws.toolkits.resources.message
+
+class FeatureDevClientUtilTest : FeatureDevTestBase() {
+    @Rule
+    @JvmField
+    val ruleChain = RuleChain(projectRule, disposableRule)
+
+    private class TestTelemetryService(
+        publisher: TelemetryPublisher = NoOpPublisher(),
+        batcher: TelemetryBatcher
+    ) : TelemetryService(publisher, batcher)
+
+    private lateinit var featureDevClient: FeatureDevClient
+    private lateinit var telemetryService: TelemetryService
+    private lateinit var batcher: TelemetryBatcher
+
+    private val uploadId = "test-upload-id"
+
+    @Before
+    override fun setup() {
+        featureDevClient = mock()
+        batcher = mock<TelemetryBatcher>()
+        telemetryService = TestTelemetryService(batcher = batcher)
+        ApplicationManager.getApplication().replaceService(TelemetryService::class.java, telemetryService, disposableRule.disposable)
```

Comment:
question: why is the telemetry service being mocked here?

Reply:
not needed, have removed